### PR TITLE
Create newton_backward_interpolation.py

### PR DESCRIPTION
**Pull request by quanta003**
### Describe your change:
Added "Newton Backward Interpolation" code 

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: arithmetic_analysis/newton_backward_interpolation.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# https://www.geeksforgeeks.org/newton-forward-backward-interpolation/
+from __future__ import annotations
+
+import math
+
+
+# for calculating u value
+def ucal(u: float, p: int) -> float:
```
[the remark this answers is not in the document]
Comment (quanta003) [reply arithmetic_analysis/newton_backward_interpolation.py:8]: u is backward_interpolation_parameter
p is no_of_values

File: arithmetic_analysis/newton_backward_interpolation.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    >>> ucal(1.2, 2)
+    3.168
+    """
+    temp = u
+    for i in range(p):
+        temp = temp * (u + i)
+    return temp
+
+
+def main() -> None:
```
[the remark this answers is not in the document]
Comment (quanta003) [reply arithmetic_analysis/newton_backward_interpolation.py:23]: doctest added

File: arithmetic_analysis/newton_backward_interpolation.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# https://www.geeksforgeeks.org/newton-forward-backward-interpolation/
+from __future__ import annotations
```
Comment (CaedenPH) [inline arithmetic_analysis/newton_backward_interpolation.py:2]: Why are you importing this? I see no typehints that aren't supported in later versions

Comment (quanta003) [reply arithmetic_analysis/newton_backward_interpolation.py:2]: Sure, I am removing it

**Comment by quanta003** [comment]
Please merge it if no issues found @spazm @CaedenPH @darkstar @bit-man

File: arithmetic_analysis/newton_backward_interpolation.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    >>> ucal(1, 2)
+    2
+    >>> ucal(1.1, 2)
+    2.5410000000000004
+    >>> ucal(1.2, 2)
+    3.168
+    """
+    temp = value_of_u
+    for i in range(value_of_n):
+        temp = temp * (value_of_u + i)
```
Comment (CaedenPH) [inline arithmetic_analysis/newton_backward_interpolation.py:17]:
```suggestion
        temp *= value_of_u + i
```

File: arithmetic_analysis/newton_backward_interpolation.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
+# https://www.geeksforgeeks.org/newton-forward-backward-interpolation/
```
Comment (CaedenPH) [inline arithmetic_analysis/newton_backward_interpolation.py:1]: Could you give a brief summary of what you're doing in the file?

File: arithmetic_analysis/newton_backward_interpolation.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+# for calculating u value
+def ucal(value_of_u: float, value_of_n: int) -> float:
+    """
+    >>> ucal(1, 2)
+    2
+    >>> ucal(1.1, 2)
+    2.5410000000000004
+    >>> ucal(1.2, 2)
+    3.168
+    """
```
Comment (CaedenPH) [inline arithmetic_analysis/newton_backward_interpolation.py:14]: Here you could add some range and type checks to ensure the type and range is correct

File: arithmetic_analysis/newton_backward_interpolation.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+
+# for calculating u value
+def ucal(value_of_u: float, value_of_n: int) -> float:
+    """
+    >>> ucal(1, 2)
+    2
+    >>> ucal(1.1, 2)
+    2.5410000000000004
+    >>> ucal(1.2, 2)
```
Comment (CaedenPH) [inline arithmetic_analysis/newton_backward_interpolation.py:12]: If you add the range and type check, you can add som doctests that error with "incorrect type", etc

File: arithmetic_analysis/newton_backward_interpolation.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
+# https://www.geeksforgeeks.org/newton-forward-backward-interpolation/
+import math
+
+
+# for calculating u value
+def ucal(value_of_u: float, value_of_n: int) -> float:
```
Comment (CaedenPH) [inline arithmetic_analysis/newton_backward_interpolation.py:6]: Is `ucal` an abbreviation for something?

File: arithmetic_analysis/newton_backward_interpolation.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
+# https://www.geeksforgeeks.org/newton-forward-backward-interpolation/
+import math
+
+
+# for calculating u value
```
Comment (CaedenPH) [inline arithmetic_analysis/newton_backward_interpolation.py:5]: Could you put this inside the docstring and explain a bit more?

File: arithmetic_analysis/newton_backward_interpolation.py [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+    for i in range(1, n):
+        for j in range(n - 1, i - 1, -1):
+            y[j][i] = y[j][i - 1] - y[j - 1][i - 1]
+
+    summ = y[n - 1][0]
+    for i in range(1, n):
+        summ += (ucal(u, i) * y[n - 1][i]) / math.factorial(i)
+
+    print(f"the value at {value} is {summ}")
```
Comment (CaedenPH) [inline arithmetic_analysis/newton_backward_interpolation.py:55]: Perhaps you could move this into a `main` function

**Comment by tianyizheng02** [comment]
Thanks for your contribution, but unfortunately there are too many issues with this PR for us to merge and you never fixed them following @CaedenPH's code review. Apart from the issues that he pointed out, your algorithmic code is almost entirely contained within a main block rather than separate functions. In addition, there should be minimal side effects, so printing should be avoided when possible. Please read the contributing guidelines if you wish to contribute.

**Pull request closed without merging**